kqueen/engines/aks.py

```diff
@@ -281,7 +283,30 @@ def cluster_get(self):
         return cluster
 
     def cluster_list(self):
-        """AKS engine don't support list of clusters"""
-        # TODO: it does, add list of clusters
-
+        """Is not needed in AKS"""
         return []
+
+    @classmethod
+    def engine_status(cls, **kwargs):
+        try:
+            credentials = ServicePrincipalCredentials(client_id=kwargs.get('client_id'),
+                                                      secret=kwargs.get('secret'),
+                                                      tenant=kwargs.get('tenant'))
+        except AuthenticationError:
+            logger.warning('Invalid credentials for {} Azure Provisioner'.format(cls.name))
+            return config.get('PROVISIONER_ERROR_STATE')
+        except Exception as e:
+            logger.warning('{} Azure Provisioner validation failed with the following reason {}'.format(cls.name,
+                                                                                                        e.message))
+            return config.get('PROVISIONER_UNKNOWN_STATE')
+        client = ContainerServiceClient(credentials, kwargs.get('subscription_id'))
+        try:
+            list(client.managed_clusters.list_by_resource_group(kwargs.get('resource_group_name')))
+        except CloudError as e:
+            logger.warning('Invalid parameters for {} Azure Provisioner: {}'.format(cls.name, e.message))
+            return config.get('PROVISIONER_ERROR_STATE')
+        except Exception as e:
```

Comment:
take a look on kqueen/engines/gce and do the similar for logging and status returning

Reply:
for ex:
```            except Exception as e:
                     msg = 'Failed to discover GCE project. Check that credentials is valid. Error:'
                     logger.exception(msg)
                     return config.get('PROVISIONER_UNKNOWN_STATE')```

Reply:
main goal - provide all engine status checks with equal arch